fix(ui): improve ui accessibility

decimozs opened the pull request:
# Pull Request: UI and Code Refinements for `AwesomePageClient` & `ProjectsPage`

This pull request enhances **code readability, consistency, and user experience** in the `AwesomePageClient` and `ProjectsPage` components. The main highlights include the introduction of a **new `formatCategory` utility function**, UI refinements, and general code cleanup.

---

## **Utility and Display Improvements**
* Added `formatCategory` in `src/lib/utils.ts` to standardize category name formatting, especially for acronyms like `SDK` and `CMS`.  
  * Now applied to the category dropdown in `AwesomePageClient`.  
  [Diff 1](diffhunk://#diff-da10bcc7d8dff796f50c9358a7268eb5abd2c50f460aecc4b2be1c691a966224L1-R20) | [Diff 2](diffhunk://#diff-33c724c91f395ccaa47b8537d55b65970ceb07b5c9488a24074ca5e290de67d7L146-R188)

---

## **UI and Code Consistency**
* Refactored JSX for better readability, including multiline props and children formatting for buttons, cards, and select items.  
  [Diff 1](diffhunk://#diff-33c724c91f395ccaa47b8537d55b65970ceb07b5c9488a24074ca5e290de67d7L105-R145) | [Diff 2](diffhunk://#diff-77154bf8a7b0126e79f2818a349df89aaaffda07cf88710602c844958985b12fL48-R63)  
* Enhanced the **“Clear filters”** button styling for better visibility and consistency.

---

## **General Refactoring**
* Improved variable assignments and filter logic in `AwesomePageClient` for readability and maintainability.  
  [Diff 1](diffhunk://#diff-33c724c91f395ccaa47b8537d55b65970ceb07b5c9488a24074ca5e290de67d7L75-R95) | [Diff 2](diffhunk://#diff-33c724c91f395ccaa47b8537d55b65970ceb07b5c9488a24074ca5e290de67d7L85-R106)

---

## **Minor UI Enhancements**
* Improved project details layout, badges, and external links for **better responsiveness and visual consistency**.  
  [Diff 1](diffhunk://#diff-33c724c91f395ccaa47b8537d55b65970ceb07b5c9488a24074ca5e290de67d7L232-R289) | [Diff 2](diffhunk://#diff-33c724c91f395ccaa47b8537d55b65970ceb07b5c9488a24074ca5e290de67d7L271-R334)

---

## **Before & After Examples**

### **Search Project**
**Before:**  
<img width="1155" height="440" alt="image" src="https://github.com/user-attachments/assets/a4728ba6-d53a-4e61-b2cc-a35a8962abe7" />  
<img width="416" height="700" alt="image" src="https://github.com/user-attachments/assets/8c819558-6998-4717-952c-c3054d3b3aac" />

**After:**  
<img width="1180" height="463" alt="image" src="https://github.com/user-attachments/assets/6ca232b3-3540-41cb-97e4-72ef48df170e" />  
<img width="398" height="694" alt="image" src="https://github.com/user-attachments/assets/2abc4bac-6a60-42bb-8516-6399d5d010b4" />

---

### **Filtering Categories**
**Before:**  
<img width="420" height="535" alt="image" src="https://github.com/user-attachments/assets/867efd64-010e-4d0f-a8e2-f58817b48d17" />  

**After:**  
<img width="441" height="539" alt="image" src="https://github.com/user-attachments/assets/30a37b0a-dc7e-4062-b36e-66aa7bc03487" />  

---

### **Project Card**
**Before:**  
<img width="1169" height="401" alt="image" src="https://github.com/user-attachments/assets/594bb956-cd0e-4d4e-9d2c-79a247656980" />  

**After:**  
<img width="1170" height="443" alt="image" src="https://github.com/user-attachments/assets/b6560305-c881-482d-a4db-602cdfb61700" />  


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added loading and error UI with retry for data fetching.
  * Added a clear-filters action to reset search and filter selections.

* **Improvements**
  * Enhanced category formatting and display logic.
  * Refined buttons, cards, filter selects, and project grid layout for better responsiveness and accessibility.
  * Updated empty-state messaging and behavior when no projects are found.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

decimozs committed:
update: fix ui accessibility

decimozs committed:
update: fix ui accessibility

jofftiquez commented:
Hi @decimozs thanks for this! Were you able to see the review items from CodeRabbit? Thank you!

decimozs committed:
fix: Icon-only external link buttons need accessible names

decimozs commented:
> Hi @decimozs thanks for this! Were you able to see the review items from CodeRabbit? Thank you!

Hi @jofftiquez yes, i did. i’ve just pushed a new commit that addresses the feedback from CodeRabbit. 

You can find the changes here: https://github.com/OSSPhilippines/ossph.org/pull/96/changes/48b0b865e8764c1f645bfb6149a6000c756c633f

jofftiquez commented:
> > Hi @decimozs thanks for this! Were you able to see the review items from CodeRabbit? Thank you!
> 
> Hi @jofftiquez yes, i did. i’ve just pushed a new commit that addresses the feedback from CodeRabbit.
> 
> You can find the changes here: [48b0b86](https://github.com/OSSPhilippines/ossph.org/commit/48b0b865e8764c1f645bfb6149a6000c756c633f)

Thanks @decimozs I've approved the vercel preview. I will check the output and merge if everything is good. Thank you!
